### PR TITLE
ac : Week 5 : Added the otherExpense table

### PR DESCRIPTION
**Pull request by its-me-Navneet**
# Issues fixed in this PR :

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
-billiard==3.6.3.0
+billiard==3.6.4.0
 Brotli==1.0.9
 bs4==0.0.1
 celery==5.0.5
 certifi==2020.12.5
 cffi==1.14.4
 chardet==4.0.0
-click==7.1.2
+click==8.0.3
```
Comment (akshatnema) [inline requirements.txt]: @its-me-Navneet kindly remove the changes of `requirements.txt` file.

**Review by akshatnema — CHANGES_REQUESTED** [review]

File: FusionIIIT/applications/finance_accounts/models.py [hunk 2]
```diff
@@ -2,11 +2,11 @@
...
-    month = models.CharField(max_length = 70 , null = True)
-    year = models.IntegerField(null = True)
-    pf = models.IntegerField(null = True)
-    name = models.CharField(max_length = 70)
-    designation = models.CharField(max_length = 50)
+    month = models.CharField(max_length=70, null=True)
+    year = models.IntegerField(null=True)
+    pf = models.IntegerField(null=True)
+    name = models.CharField(max_length=70)
+    designation = models.CharField(max_length=50)
```
Comment (PranshuNayak) [inline FusionIIIT/applications/finance_accounts/models.py]: These are indentation changes , please remove them. Don't push indentation changes

File: FusionIIIT/applications/finance_accounts/models.py [hunk 3]
```diff
@@ -25,51 +25,60 @@ class Paymentscheme(models.Model):
...
-    ass_registrar_aud_verify = models.BooleanField(default = False)
-    registrar_director_verify = models.BooleanField(default = False)
-    runpayroll = models.BooleanField(default = False)
-    view = models.BooleanField(default = True)
+    senior_verify = models.BooleanField(default=False)
+    ass_registrar_verify = models.BooleanField(default=False)
+    ass_registrar_aud_verify = models.BooleanField(default=False)
+    registrar_director_verify = models.BooleanField(default=False)
+    runpayroll = models.BooleanField(default=False)
+    view = models.BooleanField(default=True)
```
Comment (PranshuNayak) [inline FusionIIIT/applications/finance_accounts/models.py:35]: remove these indentation changes too, and keep it in its original form.

File: FusionIIIT/applications/finance_accounts/models.py [hunk 4]
```diff
@@ -25,51 +25,60 @@ class Paymentscheme(models.Model):
...
 
     class Meta:
-       constraints = [
-            models.UniqueConstraint(fields=['month', 'year', 'pf'], name='Unique Contraint 1')
+        constraints = [
+            models.UniqueConstraint(
+                fields=['month', 'year', 'pf'], name='Unique Contraint 1')
         ]
```
Comment (PranshuNayak) [inline FusionIIIT/applications/finance_accounts/models.py:40]: remove these indentation changes too

File: FusionIIIT/applications/finance_accounts/models.py [hunk 5]
```diff
@@ -25,51 +25,60 @@ class Paymentscheme(models.Model):
...
-        FromWhom = models.CharField(max_length=80)
-        Purpose = models.CharField(max_length=20)
-        Date = models.DateField()
+    receipt_id = models.AutoField(primary_key=True)
+    TransactionId = models.IntegerField(default=0, unique=True)
+    ToWhom = models.CharField(max_length=80)
+    FromWhom = models.CharField(max_length=80)
+    Purpose = models.CharField(max_length=20)
+    Date = models.DateField()
+
```
Comment (PranshuNayak) [inline FusionIIIT/applications/finance_accounts/models.py]: remove these indentation changes too

File: FusionIIIT/applications/finance_accounts/models.py [hunk 6]
```diff
@@ -25,51 +25,60 @@ class Paymentscheme(models.Model):
...
-        FromWhom = models.CharField(max_length=80)
-        Purpose = models.CharField(max_length=20)
-        Date = models.DateField()
+    payment_id = models.AutoField(primary_key=True)
+    TransactionId = models.IntegerField(default=0, unique=True)
+    ToWhom = models.CharField(max_length=80)
+    FromWhom = models.CharField(max_length=80)
+    Purpose = models.CharField(max_length=20)
+    Date = models.DateField()
+
```
Comment (PranshuNayak) [inline FusionIIIT/applications/finance_accounts/models.py]: remove these indentation changes too

**Review by PranshuNayak — COMMENTED** [review]
Please don't push indentation changes. I have pointed at some of the places but there are a lot of indentation changes that you have pushed. Kindly rectify.

File: FusionIIIT/templates/financeAndAccountsModule/hello.html [hunk 7]
```diff
@@ -0,0 +1 @@
+Print(hello)
```
Comment (PranshuNayak) [inline FusionIIIT/templates/financeAndAccountsModule/hello.html]: Don't push the unnecessary file. Also, keep the commit message concise and clear.

File: FusionIIIT/applications/finance_accounts/models.py [hunk 8]
```diff
@@ -31,11 +31,12 @@ class Paymentscheme(models.Model):
     registrar_director_verify = models.BooleanField(default=False)
     runpayroll = models.BooleanField(default=False)
     view = models.BooleanField(default=True)
+    userId = models.CharField(max_length=20, null = True)
```
Comment (PranshuNayak) [inline FusionIIIT/applications/finance_accounts/models.py:35]: you should use the get_user_model whenever linking a user to a table using the Foreign Key.

**Comment by its-me-Navneet** [comment]
@akshatnema I have undone the changes of requirements.txt.

**Comment by akshatnema** [comment]
@its-me-Navneet Kindly resolve the changes requested by the @PranshuNayak

**Comment by its-me-Navneet** [comment]
@PranshuNayak I have now remove those indentation changes.

File: FusionIIIT/applications/finance_accounts/models.py (hunk 8 above)
Comment (PranshuNayak) [reply FusionIIIT/applications/finance_accounts/models.py:35]: @its-me-Navneet

**Comment by akshatnema** [comment]
@its-me-Navneet as your module reviewer to review the PR

File: FusionIIIT/applications/finance_accounts/models.py (hunk 8 above)
Comment (its-me-Navneet) [reply FusionIIIT/applications/finance_accounts/models.py:35]: @PranshuNayak  Now we dont need userid we are doing it through pf number which is unique for every employee and thus we have made the changes.

**Review by BhavyaKollapaneni2003 — APPROVED** [review]
Approve @akshatnema

**Comment by PranshuNayak** [comment]
@samay-rgb

**Comment by PranshuNayak** [comment]
@its-me-Navneet push the migrations for newly created db changes only.

File: FusionIIIT/applications/finance_accounts/models.py (hunk 8 above)
Comment (PranshuNayak) [reply FusionIIIT/applications/finance_accounts/models.py:35]: If you have not made any changes in the models.py file remove it.

File: FusionIIIT/applications/finance_accounts/render.py [hunk 9]
```diff
@@ -1,22 +1,63 @@
...
+
+    #     response = HttpResponse(content_type='application/pdf')
+    #     response['Content-Disposition'] = 'attachment; filename="salary.pdf"'
+    #     # create a pdf
+    #     pisa_status = pisa.CreatePDF(
+    #         html, dest=response)
+    #     # if error then show some funy view
+    #     if pisa_status.err:
+    #         return HttpResponse('We had some errors <pre>' + html + '</pre>')
+    #     return response
```
Comment (akshatnema) [inline FusionIIIT/applications/finance_accounts/render.py:25]: Kindly remove this comment if not needed.

**Pull request merged**